### PR TITLE
for Issue #268 - channelValid test

### PR DESCRIPTION
**Pull request by estefysc**
Added a unit test for the channelValid function.

Screenshot 2023-05-01 at 5 41 58 PM

### CODE REVIEW

File: backend/api/chatLink/utils/validateChannel.test.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    });
+
+    // test if channel is provided otherwise throw new Error("channel - required param");
+    it('should throw an error is the channel is not provided', async () => {
+        await expect(channelValid('')).rejects.toThrow("channel - required param");
+    });
+
+    it('should return NOT_FOUND if the channel does not exist in the db', async () => {
+        const findOneFromDBMock = db.findOneFromDB as jest.MockedFunction<typeof db.findOneFromDB>;
+        findOneFromDBMock.mockResolvedValueOnce(null);
```
Comment (muke1908) [inline backend/api/chatLink/utils/validateChannel.test.ts:18]: `db.findOneFromDB = jest.fn().mockResolvedValueOnce(null);`
this should work too.

File: backend/api/chatLink/utils/validateChannel.test.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        // WHere is the above being used?
+        const result = await channelValid('nonexistent_channel');
+        expect(result.valid).toBe(false);
+        expect(result.state).toBe(CHANNEL_STATE.NOT_FOUND);
```
Comment (muke1908) [inline backend/api/chatLink/utils/validateChannel.test.ts:22]: One more assertion that can be useful: 
`expect(db.findOneFromDB).toBeCalledWith({ hash: 'nonexistent_channel' });`

File: backend/api/chatLink/utils/validateChannel.test.ts (hunk 1 above)
Comment (estefysc) [reply backend/api/chatLink/utils/validateChannel.test.ts:18]: Thank you, I'll make the change.

File: backend/api/chatLink/utils/validateChannel.test.ts (hunk 2 above)
Comment (estefysc) [reply backend/api/chatLink/utils/validateChannel.test.ts:22]: Thanks for the review, will work on this.

**Pull request merged**